test/html/pipeline/mention_filter_test.rb

```diff
@@ -72,6 +72,20 @@ def test_links_to_more_info_when_info_url_given
       filter(body, '/', 'https://github.com/blog/821').to_html
   end
 
+  def test_base_url_tilde_with_slash
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"~jch\" class=\"user-mention\">@jch</a>"
```

Comment:
You are so right! Thank you for reviewing! Just updated, please take a look again! :blush: